app/src/lib/editors/darwin.ts

```diff
@@ -0,0 +1,81 @@
+import * as Fs from 'fs'
+import * as Path from 'path'
+
+/**
+ * appPath will raise an error if it cannot find the program.
+ */
+const appPath: (bundleId: string) => Promise<string> = require('app-path')
+
+import {
+  FoundEditor,
+  AtomLabel,
+  VisualStudioCodeLabel,
+  SublimeTextLabel,
+} from './shared'
+
+function pathExists(path: string): Promise<boolean> {
+  return new Promise<boolean>((resolve, reject) => {
+    Fs.exists(path, exists => {
+      resolve(exists)
+    })
+  })
+}
+
+/**
+ * Lookup the known external editors using the bundle ID that each uses
+ * to register itself on a user's machine.
+ */
+export async function getAvailableEditors(): Promise<
+  ReadonlyArray<FoundEditor>
+> {
+  const results: Array<FoundEditor> = []
+
+  try {
+    // invoked without assigning the result so we verify the app is installed
+    await appPath('com.github.atom')
```

Comment:
Doing this verification in two steps feels a bit paranoid, would love to hear other people's thoughts about validating an app is installed (and the one we want).